### PR TITLE
fix: Restrict Azure state logging to WARNING and above

### LINKED ISSUE #7723: Azure State backend - silence logs
### Feature scope

Configuration (settings parsing, validation, etc.)

### Description

Using Azure as a backend fills the meltano log with noise, multiple log lines for each http request, basically making the log unusable.

So far I have managed to silence these logs with an addition to `meltano/core/state_store/azure.py`

```
import logging

logger=logging.getLogger('azure.core.pipeline.policies.http_logging_policy')
logger.setLevel(logging.WARNING)
```

Without the suppression above, the logs are full of multiline messages like this, logs become virtually useless:

```
2023-05-31T08:29:59.892347Z [info     ] Request URL: 'https://***.blob.core.windows.net/meltano-state/dev%3Atap-freshdesk-to-target-postgres/lock'
Request method: 'HEAD'
Request headers:
    'x-ms-version': 'REDACTED'
    'Accept': 'application/xml'
    'User-Agent': 'azsdk-python-storage-blob/12.16.0 Python/3.9.16 (Linux-5.10.102.2-microsoft-standard-x86_64-with-glibc2.31)'
    'x-ms-date': 'REDACTED'
    'x-ms-client-request-id': '54f13080-ff8d-11ed-8616-00155da6d400'
    'Authorization': 'REDACTED'
No body was attached to the request
2023-05-31T08:29:59.901579Z [info     ] Response status: 404
Response headers:
    'Transfer-Encoding': 'chunked'
    'Server': 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0'
    'x-ms-request-id': 'fe209118-f01e-0044-229a-935122000000'
    'x-ms-client-request-id': '54f13080-ff8d-11ed-8616-00155da6d400'
    'x-ms-version': 'REDACTED'
    'x-ms-error-code': 'BlobNotFound'
    'Date': 'Wed, 31 May 2023 08:29:59 GMT'
2023-05-31T08:29:59.913150Z [info     ] Request URL: 'https://***.blob.core.windows.net/meltano-state/dev%3Atap-freshdesk-to-target-postgres/lock?comp=REDACTED&blockid=REDACTED'
Request method: 'PUT'
Request headers:
    'Content-Length': '17'
    'x-ms-version': 'REDACTED'
    'Content-Type': 'application/octet-stream'
    'Accept': 'application/xml'
    'User-Agent': 'azsdk-python-storage-blob/12.16.0 Python/3.9.16 (Linux-5.10.102.2-microsoft-standard-x86_64-with-glibc2.31)'
    'x-ms-date': 'REDACTED'
    'x-ms-client-request-id': '54f45b48-ff8d-11ed-8616-00155da6d400'
    'Authorization': 'REDACTED'
A body is sent with the request
```

Would you consider a PR to suppress the INFO logs from Azure Blob Client?

### PR DESCRIPTION
**Pull request by acarter24**
Closes #7723

### CODE REVIEW

File: src/meltano/core/logging/utils.py [hunk 1]
```diff
@@ -121,6 +121,11 @@ def default_config(log_level: str) -> dict:
                 "level": logging.INFO,
                 "propagate": False,
             },
+            "azure.core.pipeline.policies.http_logging_policy": {  # see #7723 , azure http logs at info level, too noisy
```
Comment (WillDaSilva) [inline src/meltano/core/logging/utils.py:124]:
```suggestion
            # Azure HTTP logs at info level are too noisy; see
            # https://github.com/meltano/meltano/issues/7723
            "azure.core.pipeline.policies.http_logging_policy": {
```

File: src/meltano/core/logging/utils.py [hunk 2]
```diff
@@ -121,6 +121,13 @@ def default_config(log_level: str) -> dict:
                 "level": logging.INFO,
                 "propagate": False,
             },
+            # Azure HTTP logs at info level are too noisy; see
+            # https://github.com/meltano/meltano/issues/7723
+            "azure.core.pipeline.policies.http_logging_policy": {
+                "handlers": ["console"],
+                "level": logging.WARNING,
+                "propagate": False,
+            },
```
Comment (edgarrmondragon) [inline src/meltano/core/logging/utils.py:130]: I think we can also achieve the desired behavior if we omit `handlers` and enable `propagate`

```suggestion
                "level": logging.WARNING,
                "propagate": True,
            },
```

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]

**Comment by edgarrmondragon** [comment]
@acarter24 I can't push to your fork's `main` branch but this changes seem to work fine:

```diff
                 "propagate": True,
             },
             "snowplow_tracker.emitters": {
-                "handlers": ["console"],
                 "level": logging.ERROR,
             },
             "urllib3": {
-                "handlers": ["console"],
                 "level": logging.INFO,
-                "propagate": False,
             },
             "asyncio": {
-                "handlers": ["console"],
                 "level": logging.INFO,
-                "propagate": False,
             },
             # Azure HTTP logs at info level are too noisy; see
             # https://github.com/meltano/meltano/issues/7723
             "azure.core.pipeline.policies.http_logging_policy": {
-                "handlers": ["console"],
                 "level": logging.WARNING,
-                "propagate": False,
             },
         },
     }
```

**Comment by WillDaSilva** [comment]
Thanks for the updates @acarter24

@edgarrmondragon is on vacation, and will be back on the 19th or 20th.

He knows this PR and code better than I, but if you feel this is sufficiently urgent (i.e. it's causing problems for you and you would like the fix to be released in a new version of Meltano sooner rather than later) I can review this today or tomorrow. What would you prefer @acarter24?

**Review by edgarrmondragon — APPROVED** [review]
Thanks @acarter24!

**Pull request merged**